src/processes/news-process.js

```diff
@@ -1,213 +1,250 @@
-
-            const site_name = data.site_name
-            const activity = {
-                domain: {
-                    name, site_name, info, image: logo, domain_page_id: data.domain_page_id
-                },
-                content: { ...data, ...dateCreted }
-            }
-
-            await putMainFeed(job, name, logo, created_domain, data, findNewsLink?.post_id);
-            await new ElasticNewsLink().putToIndexFromGetstreamObject(findNewsLink)
-            // await postToGetstream(activity, job.user_id);
-            message = 'url news not unique'
-        } else {
-
-            data.news_link_id = uuidv4();
-            data.url = data.news_url;
-
-            const site_name = data.site_name
-            const activity = {
-                domain: {
-                    name, site_name, info, image: logo, domain_page_id: data.domain_page_id
-                },
-                content: { ...data, ...dateCreted }
-            }
-
-            let result = await postToGetstream(activity, job.user_id);
-            await putMainFeed(job, name, logo, created_domain, data, result?.id);
-
-            let postId = result.returnActivityId
-            console.info(result);
-            console.info(`postId in Process : ${postId}`);
-
-            let newsLink = await NewsLink.create({ ...data, ...dateCreted, post_id: result.id })
-            await new ElasticNewsLink().putToIndexFromGetstreamObject(newsLink)
-            message = 'news link created'
+  try {
+    const findNewsLink = await NewsLink.findOne({
+      where: {news_url: data.news_url}
+    });
+    let message;
+    if (findNewsLink) {
+      console.info('id', findNewsLink.news_link_id);
+      data.news_link_id = findNewsLink.news_link_id;
+      data.url = data.news_url;
+
+      const {site_name} = data;
+      const activity = {
+        domain: {
+          name,
+          site_name,
+          info,
+          image: logo,
+          domain_page_id: data.domain_page_id
+        },
+        content: {
+          ...data,
+          ...dateCreted,
+          content_published_at: findNewsLink?.created_article || data?.content_published_at
         }
-
-        return message
-    } catch (error) {
-        return error
+      };
+
+      await putMainFeed(job, name, logo, created_domain, data, findNewsLink?.post_id);
+      await new ElasticNewsLink().putToIndexFromGetstreamObject(findNewsLink);
+      // await postToGetstream(activity, job.user_id);
+      message = 'url news not unique';
+    } else {
+      data.news_link_id = uuidv4();
+      data.url = data.news_url;
+
+      const {site_name} = data;
+      const activity = {
+        domain: {
+          name,
+          site_name,
+          info,
+          image: logo,
+          domain_page_id: data.domain_page_id
+        },
+        content: {
+          ...data,
+          ...dateCreted,
+          content_published_at: findNewsLink?.created_article || data?.content_published_at
+        }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Similar to the previous comment, the optional chaining operator `?.` is used here. It's unnecessary because if `findNewsLink` or `data` is undefined, there would be a larger issue at hand. Also, the order of precedence seems incorrect. The new value `data?.content_published_at` should take precedence over the old value `findNewsLink?.created_article`.
```diff
-          content_published_at: findNewsLink?.created_article || data?.content_published_at
+          content_published_at: data.content_published_at || findNewsLink.created_article
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->